### PR TITLE
✨ Add Login Page Function

### LINKED ISSUE #43: Login Function
- [x] Check user data is available
- [x] forward to the start page if available

### PR DESCRIPTION
**Pull request by dhnazarenko**
fixes #43

### CODE REVIEW

**Review by Dev-dfm — APPROVED** [review]
lgtm 🚀

File: src/app/pages/Login/Login.tsx [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
 
-  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
     event.preventDefault();
-    alert(`Login with ${username}:${password}`);
+
+    try {
+      const user: Partial<User> = { username, password };
+      await postLoginUser(user);
+      history.push('/register');
```
Comment (lmachens) [inline src/app/pages/Login/Login.tsx]: Why should you be redirected to `/register` after a successful login?

File: src/server/routes.ts [hunk 2]
```diff
@@ -13,4 +13,19 @@ router.get('/users', async (_req, res) => {
...
 
+router.post('/users/login', async (req, res, next) => {
+  try {
+    const { username, password } = req.body;
+    const user = await readUser({ username, password });
+    if (!user) {
+      res.status(404).send('User not found');
+      return;
+    }
+
```
Comment (lmachens) [inline src/server/routes.ts:24]: It would be awesome to send a cookie to the client to identify the user in upcoming requests

**Review by lmachens — COMMENTED** [review]
gj

File: src/app/pages/Login/Login.tsx (hunk 1 above)
Comment (dhnazarenko) [reply src/app/pages/Login/Login.tsx]: Thanks! That was not planned.

File: src/server/routes.ts (hunk 2 above)
Comment (dhnazarenko) [reply src/server/routes.ts:24]: You're right. Is done.

**Pull request merged**